pymine_net/net/socket/server.py

```diff
@@ -0,0 +1,65 @@
+import socket
+import threading
+from typing import Dict, List, Tuple, Union
+
+from pymine_net.net.server import AbstractProtocolServer, AbstractProtocolServerClient
+from pymine_net.net.socket.stream import SocketTCPStream
+from pymine_net.strict_abc import abstract
+from pymine_net.types.packet import ClientBoundPacket, ServerBoundPacket
+from pymine_net.types.packet_map import PacketMap
+
+
+class SocketProtocolServerClient(AbstractProtocolServerClient):
+    def __init__(self, stream: SocketTCPStream, packet_map: PacketMap):
+        super().__init__(stream, packet_map)
+        self.stream = stream  # redefine this cause typehints
+
+    def read_packet(self) -> ServerBoundPacket:
+        length = self.stream.read_varint()
+        return self._decode_packet(self.stream.read(length))
+
+    def write_packet(self, packet: ClientBoundPacket) -> None:
+        self.stream.write(self._encode_packet(packet))
+
+
+class SocketProtocolServer(AbstractProtocolServer):
+    def __init__(self, host: str, port: int, protocol: Union[int, str], packet_map: PacketMap):
+        super().__init__(host, port, protocol, packet_map)
+
+        self.connected_clients: Dict[Tuple[str, int], SocketProtocolServerClient] = {}
+
+        self.sock: socket.socket = None
+        self.threads: List[threading.Thread] = []
+        self.running = False
+
+    def run(self) -> None:
+        self.running = True
+
+        with socket.socket(socket.AF_INET, socket.SOCK_STREAM) as self.sock:
```

Comment:
This forces the only way to listen to be host/port right?
Could other uses want other types of connection,
 like Unix stream sockets files
(Or whatever)

That could be used for non privelidged fowardering, over a tunnel or whatever

Or are you forced to create your own implementation

Reply:
Yeah it does force that, but it's not hard to make your own implementation due to how abstracted away everything is.